tests/63-live-notify_addfd.c

```diff
@@ -0,0 +1,210 @@
+/**
+ * Seccomp Library test program
+ *
+ * Copyright (c) 2025 Microsoft Corporation <sudpandit@microsoft.com>
+ * Author: Sudipta Pandit <sudpandit@microsoft.com>
+ */
+
+/*
+ * This library is free software; you can redistribute it and/or modify it
+ * under the terms of version 2.1 of the GNU Lesser General Public License as
+ * published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, see <http://www.gnu.org/licenses>.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <seccomp.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+
+int send_fd(int sock, int fd)
+{
+	struct iovec iov = {.iov_base = "F", .iov_len = 1};
+	char buffer[CMSG_SPACE(sizeof(fd))]; // Do i need to set it to zero?
+	memset(buffer, 0, sizeof(buffer));
+
+	struct msghdr msg = {
+		.msg_iov = &iov,
+		.msg_iovlen = 1,
+		.msg_control = buffer,
+		.msg_controllen = sizeof(buffer)
+	};
+
+	struct cmsghdr *cmsg = CMSG_FIRSTHDR(&msg);
+	cmsg->cmsg_level = SOL_SOCKET;
+	cmsg->cmsg_type = SCM_RIGHTS;
+	cmsg->cmsg_len = CMSG_LEN(sizeof(fd));
+
+	memcpy(CMSG_DATA(cmsg), &fd, sizeof(fd));
+
+	return sendmsg(sock, &msg, 0);
+}
+
+int recv_fd(int sock)
+{
+	char m_buffer[1];
+	struct iovec iov = {.iov_base = m_buffer, .iov_len = 1};
+
+	char buffer[CMSG_SPACE(sizeof(int))];
+	struct msghdr msg = {
+		.msg_iov = &iov,
+		.msg_iovlen = 1,
+		.msg_control = buffer,
+		.msg_controllen = sizeof(buffer)
+	};
+	struct cmsghdr *cmsg = CMSG_FIRSTHDR(&msg);
+
+	if (recvmsg(sock, &msg, 0) < 0)
+		return -1;
+
+	int fd;
```

Comment:
To avoid sounding like a broken record, I'll only make this comment once more.

You should probably declare variables at the top of the function (even though this is no longer required) for two reasons:
* It matches the existing libseccomp coding style
* It's much easier for the reader to know how this function is allocating variables and affecting the stack

Reply:
Got it thanks for pointing this out, I should have been mindful about the current coding style and conventions.